### PR TITLE
Feat/add monitoring

### PR DESCRIPTION
**Pull request by sandromello**
- Add sentry to client/agent
- Add [pyroscope for continuous profiling](https://pyroscope.io/)

## Sentry

The agent will receive a DSN in runtime to capture errors, we could later on change it configuring the env var `AGENT_SENTRY_DSN`

The client has a hardcoded DSN, it reports errors in synchronous due to the characteristic of how the client works.

## Profiling

This PR introduces profiling of agent/gateway, this is an experimental change to add more observability to these components. Having profiling will help us to diagnose problems more precise and understand where to look for issues.

Pyroscope has features such as:

- Mutex monitoring (we had a recent problem with that!)
- Comparing / Diffing profiles to see if we introduce bugs
- Memory/CPU Profiling explain how much each object is allocating

The cloud solution is free for 100GB / monthly (14 days retention), after that it starts at U$$500/month for 1TB of ingestion.

I've tested the opensource solution and deployed on Kubernetes, it worked very well, we could host an instance if the cost is too high.

Gateway Profiling:

image

Comparing Profiles:
image

Agent Profiling:
image

## Introduced environments

- AGENT_SENTRY_DSN
- PYROSCOPE_AUTH_TOKEN
- PYROSCOPE_INGEST_URL

If these environment variables are not set, the gateway/agent will not start the profiler or sentry

### CODE REVIEW

File: client/cmd/login.go [hunk 1]
```diff
@@ -27,11 +29,13 @@ const (
...
-	Use:   "login",
-	Short: "Authenticate at Hoop",
-	Long:  `Login to gain access to hoop usage.`,
+	Use:    "login",
+	Short:  "Authenticate at Hoop",
+	Long:   `Login to gain access to hoop usage.`,
+	PreRun: monitoring.SentryPreRun,
 	Run: func(cmd *cobra.Command, args []string) {
 		if err := doLogin(args); err != nil {
+			sentry.CaptureException(fmt.Errorf("login - failed to authenticate, err=%v", err))
```
Comment (danielbonilha) [inline client/cmd/login.go]: Maybe this is noise we should not capture?

File: gateway/transport/agent.go [hunk 2]
```diff
@@ -126,7 +148,7 @@ func (s *Server) listenAgentMessages(config plugin.Config, ag *agent.Agent, stre
 			log.Printf("received error from agent, err=%v", err)
 			return err
 		}
-		if pkt.Type == pbgateway.KeepAlive {
+		if pkt.Type == pbgateway.KeepAlive || pkt.Type == "KeepAlive" {
```
Comment (danielbonilha) [inline gateway/transport/agent.go:151]: I didnt get this.

Comment (sandromello) [reply gateway/transport/agent.go:151]: Old clients, rs

File: client/cmd/login.go (hunk 1 above)
Comment (sandromello) [reply client/cmd/login.go]: Yeah, I'm removing it

**Pull request merged**